lib/response.js

```diff
@@ -717,9 +717,14 @@ res.header = function header(field, val) {
       : String(val);
 
     // add charset to content-type
-    if (field.toLowerCase() === 'content-type' && !charsetRegExp.test(value)) {
-      var charset = mime.charsets.lookup(value.split(';')[0]);
-      if (charset) value += '; charset=' + charset.toLowerCase();
+    if (field.toLowerCase() === 'content-type') {
+      if (Array.isArray(value)) {
```

Comment:
It should probably throw for any non-string value, not just arrays. Otherwise, passing in `42` will result in the same `.split` error, so may as well just fix it completely in this pass.